src/types.rs

```diff
@@ -72,11 +75,22 @@ impl Deref for HWIPartiallySignedTransaction {
     }
 }
 
-// TODO: use Descriptors
+pub trait ToDescriptor {}
+impl ToDescriptor for String {}
+impl ToDescriptor for &String {}
+impl ToDescriptor for &str {}
```

Comment:
Sorry for nitpicking, I think those two aren't needed, and also &Descriptor<DescriptorPublicKey>. In `display_address_with_desc` you can just accept `&T` instead of `T`, and it should be fine.

```diff
diff --git a/src/interface.rs b/src/interface.rs
index e39a8a5..3bfb74a 100644
--- a/src/interface.rs
+++ b/src/interface.rs
@@ -321,7 +321,7 @@ impl HWIClient {
     }
 
     /// Returns an address given a descriptor.
-    pub fn display_address_with_desc<T>(&self, descriptor: T) -> Result<HWIAddress, Error>
+    pub fn display_address_with_desc<T>(&self, descriptor: &T) -> Result<HWIAddress, Error>
     where
         T: ToDescriptor + ToString,
     {
diff --git a/src/types.rs b/src/types.rs
index 575f2d7..55e85c7 100644
--- a/src/types.rs
+++ b/src/types.rs
@@ -77,12 +77,8 @@ impl Deref for HWIPartiallySignedTransaction {
 
 pub trait ToDescriptor {}
 impl ToDescriptor for String {}
-impl ToDescriptor for &String {}
-impl ToDescriptor for &str {}
 #[cfg(feature = "use-miniscript")]
 impl ToDescriptor for Descriptor<DescriptorPublicKey> {}
-#[cfg(feature = "use-miniscript")]
-impl ToDescriptor for &Descriptor<DescriptorPublicKey> {}
 
 #[derive(Clone, Eq, PartialEq, Debug, Deserialize)]
 pub struct HWIDescriptor<T>
 ```